smarts/core/vehicle_index.py

```diff
@@ -280,10 +280,12 @@ def vehicleitems(self) -> Iterator[Tuple[str, Vehicle]]:
         """A list of all vehicle IDs paired with their vehicle."""
         return map(lambda x: (self._2id_to_id[x[0]], x[1]), self._vehicles.items())
 
-    def vehicle_by_id(self, vehicle_id: str):
+    def vehicle_by_id(self, vehicle_id, default=...):
```

Comment:
Vehicle id is still a string, this is just a conflict with the new develop. The reason why I used `Ellipsis` here is because `None` is a common value being used which could indicate an error. I want an exact value to be specified.

Reply:
ok, but are you sure about `vehicle_id` stil being marked as a `str` on your branch?  (It's marked right now on `develop`, but when I pull your branch it's not.)